lib/node.rb

```diff
@@ -0,0 +1,38 @@
+require 'byebug'
+
+class Node
+  def add(array_chars, tree)
+    array_chars.each do |char|
+      tree[char] ||= {}
+      tree[char][:last_char] = true if array_chars[-1] == char
+      tree = tree[char]
+    end
+  end
+
+  def nodes_present?(array_chars)
+    value_char = $prefix_tree.dig(*array_chars)
+    last?(value_char)
+  end
+
+  def knit_nodes(tree, array_chars, array_words)
```

Comment:
This should be moved to Tree class, it is not logic for just node, because Node is just 1 symbol